### PR TITLE
Fix comments for state callbacks

### PR DESCRIPTION
**Pull request by gbiggs**
Fix the leader comments for two of the state callbacks, which used the wrong state names.

### CODE REVIEW

File: lifecycle/src/lifecycle_talker.cpp [hunk 1]
```diff
@@ -195,7 +195,7 @@ class LifecycleTalker : public rclcpp_lifecycle::LifecycleNode
     return lifecycle_msgs::msg::Transition::TRANSITION_CALLBACK_SUCCESS;
   }
 
-  /// Transition callback for state deactivating
+  /// Transition callback for state cleanup
```
Comment (mikaelarguedas) [inline lifecycle/src/lifecycle_talker.cpp]: should this be "cleaningup" instead of "cleanup" ?

Comment (wjwwood) [reply lifecycle/src/lifecycle_talker.cpp]: I think cleanup reads better myself.

Comment (mikaelarguedas) [reply lifecycle/src/lifecycle_talker.cpp]: I agree with the fact that it reads better. I figured we wanted to match the name of the transition state defined in the [design document](http://design.ros2.org/articles/node_lifecycle.html) and [in the same docblock below](https://github.com/ros2/demos/pull/221/files#diff-a333d83f31412712a00d05e83145ebb0R201) but am fine with both

Comment (wjwwood) [reply lifecycle/src/lifecycle_talker.cpp]: Ok, yeah for that it does make sense. It is the "cleaning up" state not the "cleanup of state".

Comment (gbiggs) [reply lifecycle/src/lifecycle_talker.cpp]: So what's the conclusion? The callback name and the state name should probably match, and ideally they would also match the design document.

Comment (mikaelarguedas) [reply lifecycle/src/lifecycle_talker.cpp]: :+1: yes the conclusion is to use "cleaningup" to match the design doc and the rest of the code

Comment (gbiggs) [reply lifecycle/src/lifecycle_talker.cpp]: OK, done. :+1: 

I think the callback name in the demo and the API should also be changed to `on_cleaningup`, but that's a separate issue.

Comment (mikaelarguedas) [reply lifecycle/src/lifecycle_talker.cpp]: >  think the callback name in the demo and the API should also be changed to on_cleaningup, but that's a separate issue.

I'm going to merge this now but feel free to open a PR to address those.

Comment (wjwwood) [reply lifecycle/src/lifecycle_talker.cpp]: > I think the callback name in the demo and the API should also be changed to on_cleaningup, but that's a separate issue.

Actually based on the article, I'd say the callback name is right, no? It's this:

capture

State is "CleaningUp" but the transition is called "onCleanup".

Comment (mikaelarguedas) [reply lifecycle/src/lifecycle_talker.cpp]: :+1:  reading the design doc I have the same reading as @wjwwood

Comment (gbiggs) [reply lifecycle/src/lifecycle_talker.cpp]: I'm happy with that.

**Pull request merged**